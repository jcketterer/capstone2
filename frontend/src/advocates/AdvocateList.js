import React, { useEffect, useState, useContext, useCallback } from 'react';
import { Redirect } from 'react-router-dom';
import UserContext from '../UserContext';
import AdvocateAPI from '../api';
import { Link } from 'react-router-dom';
import AdvocateCard from './AdvocateCard';
import SearchBar from '../SearchBar';

const AdvocateList = () => {
  const user = useContext(UserContext);
  const [advocates, setAdvocates] = useState([]);
  const [filter, setFilter] = useState('');

  useEffect(() => {
    let isRendered = true;

    async function getAdvocates(filterString = '') {
      try {
        if (isRendered && user.username) {
          let advocateRes = await AdvocateAPI.getAdvocates(filterString);
          setAdvocates(advocateRes);
        }
      } catch (err) {
        console.log(err);
      }
    }
    getAdvocates(filter);

    return () => {
      isRendered = false;
    };
  }, [filter, user.username]);

  if (!user.username) {
    return <Redirect to="/" />;
  }

  return (
    <div className="AdvocateList">
      <div className="container">
        <div className="d-inline-flex flex-column">
          <SearchBar onSubmit={setFilter} />
          {advocates.length > 0 ? (
            advocates.map(advocate => (
              <AdvocateCard advocate={advocate} key={advocate.advocateId} />
            ))
          ) : (
            <p>Advocate Not Found</p>
          )}
        </div>
      </div>
    </div>
  );
};

export default AdvocateList;
