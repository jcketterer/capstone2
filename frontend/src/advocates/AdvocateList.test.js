import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import AdvocateList from './AdvocateList';
import UserContext from '../UserContext';

it('should render without breaking', () => {
  render(
    <MemoryRouter>
      <UserContext.Provider value={{ username: 'test' }}>
        <AdvocateList />
      </UserContext.Provider>
    </MemoryRouter>
  );
});

it('should render and match snapshot', () => {
  const { asFragment } = render(
    <MemoryRouter>
      <UserContext.Provider value={{ username: 'test' }}>
        <AdvocateList />
      </UserContext.Provider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
