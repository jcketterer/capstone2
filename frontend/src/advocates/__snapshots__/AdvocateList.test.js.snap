// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render and match snapshot 1`] = `
<DocumentFragment>
  <div
    class="AdvocateList"
  >
    <div
      class="container"
    >
      <h3
        class="display-3 mb-3"
      >
        Advocates:
      </h3>
      <div
        class="d-inline-flex flex-column"
      >
        <p>
          Advocate Not Found
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
